### DISCUSSION TITLE
Kde wayland session won't login

### DISCUSSION THREAD

**Discussion by Spaceboy60** [SpiralLinux/SpiralLinux-project#491] [General]
Get a black screen, then just backs out to the display manager.  X11 KDE works without issues

I've tried installing in 2 VMs so far. Vmware and Virt-manager/kvm but same issue on both.

`journalctl -xb`

[https://pastecode.io/s/v4avqv2y](https://pastecode.io/s/v4avqv2y)

Any suggestions?

**Comment by geckolinux** [comment]
Hi there, I don't know if Wayland would normally work or not with that configuration in Debian 12, but as far as the SpiralLinux out-of-the-box configuration there is no support for Wayland beyond what you might be able to configure yourself using Debian instructions and additional packages.

**Reply by Spaceboy60** [reply]
So Wayland isn't officially supported as yet on spiral? Not particularly bothered at the minute  X11 works just fine

**Reply by geckolinux** [reply]
I don't and won't support Wayland as long as it's possible to avoid it. But Debian may or may not support it depending on the desktop environment and hardware, and SpiralLinux users that want it are welcome to configure it with Debian resources.

**Reply by AlphaElwedritsch** [reply]
what is the basis for your rigorous aversion to wayland, if I may ask?

**Reply by Spaceboy60** [reply]
> what is the basis for your rigorous aversion to wayland, if I may ask?

No answer still, must be a good reason I'm guessing.

**Reply by geckolinux** [reply]
@AlphaElwedritsch The Wayland protocol was created to serve the interests of some developers, not end users. It has created a fragmented and inconsistent sub-ecosystem inside the larger Linux ecosystem that is already fragmented and inconsistent. Additionally, the organizational structure of the Wayland project feels more like a political party than a software development team. Just look at this mess:
- https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/GOVERNANCE.md
- https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/339

And remember that this entire messy process governs just a *protocol* or *specification* and doesn't even produce any usable software. The onus is on developers of actual applications and desktop environments to actually make it work. This means that many projects are dedicating unbelievable amounts of resources to re-engineering their entire codebase to make it work with Wayland on the backend instead of spending those resources on bugs and features that actually affect how end users interact with the software. And that's in the best-case scenario where they actually achieve feature parity with how the software worked on X11, more often than not they end up with more new bugs and inconsistencies or even less features.

I despise the disingenuous premise often used to justify Wayland's existence: "Because X11 is not secure and allows for any application to spy on another application". To which I would reply: "So stop using apps that spy on other apps." Seriously, if somebody managed to install a keylogger on your system without you knowing about it then you have much bigger security problems to worry about than the architecture of your display manager.

Here are some more interesting reading materials:
- https://www.reddit.com/r/linux/comments/wwsiaf/writing_a_wayland_compositor_is_much_harder_than
     - https://www.reddit.com/r/linux/comments/wwsiaf/writing_a_wayland_compositor_is_much_harder_than/ilpgr2w
- https://www.reddit.com/r/linuxquestions/comments/1089ctd/so_what_exactly_is_wayland/jskeml7/
- https://grimoire.carcano.ch/blog/wayland-tutorial-a-wayland-howto/#The_Wayland_protocols

In summary, I think that this person's take on the situation is the most reasonable:
https://discussion.fedoraproject.org/t/why-wayland-anyways/95895/7

**Reply by ghost** [reply]
I myself would need to read the wayland protocol spec but I do not feel security and usability/convienence are mutually exclusive ... people need to stop thinking about how a given item is attacked and think about usability but maybe I just been under a rock too long.

**Comment by ghost** [comment]
I have Wayland working, It was a while ago but I will see if I can find the instructions I used  ><   so I will be back

**Comment by ghost** [comment]
And yes, I use Wayland for Android APK apps...was tired of using fat ass Fedora   : P

**Comment by ghost** [comment]
I used these instructions: 
https://docs.waydro.id/usage/install-on-desktops 
may need to add a package, not sure but I just did it to LastOS and it worked fine without adding packages, so your mileage may vary

**Comment by Spaceboy60** [comment]
I presume I follow the Ubuntu instructions since no debian instructions??

**Comment by ghost** [comment]
Debian IS Ubuntu :P   where do you think Ubuntu came from?

They are the same, but Debian is older and to a certain degree better - but
thats my opinion   lol

**Reply by Spaceboy60** [reply]
Shouldn't that be Ubuntu is DEBIAN (Opposite way round, cause if no Debian there'd be no Ubuntu :)

**Comment by ghost** [comment]
I'm not speaking for site admin but a few years (Im not sure cuz of my MS)
linuxMint (ie Ubuntu) had a Wayland issue but I think those were resolved.

To a certain degree SpiralLinux is slower than Ubuntu :P   Ubuntu was spun
off from Debian because Debian is very methodlogical about releases and was
considered too slow/too safe.  So Ubuntu was created.

I have found there are a lot of distros because people want things but like
little kids most of those people do not know they dont understand enough to
know what it is they are missing.

So, me?  I just try to stay out of it and do what I want with whatever
distro I want ... I guess I am a linux junkie just looking for a fix   lol
